lib/ContextModuleFactory.js

```diff
@@ -275,9 +276,14 @@ module.exports = class ContextModuleFactory extends ModuleFactory {
 			category
 		} = options;
 		if (!regExp || !resource) return callback(null, []);
-
+		let visited = new Set();
 		const addDirectory = (directory, callback) => {
+			const realPath = path.resolve(directory);
```

Comment:
Why we need `resolve` this? We already known `context`, so you can add `path.join(context, directory)`

Reply:
do you mean `const realPath = path.join(resource, directory)` ? as I don't think we're directly passing `context` here.

Reply:
We know context of request, so we can just join them

Reply:
But context is undefined here, can you help me figure on how to extract the context?, Is context the resource property of options here?.

Reply:
Maybe this https://github.com/webpack/webpack/blob/d5d7f3d57692fe4455c85789ae1b1a3479bb89a4/lib/ContextModuleFactory.js#L316 or https://github.com/webpack/webpack/blob/d5d7f3d57692fe4455c85789ae1b1a3479bb89a4/lib/ContextModuleFactory.js#L87, need test

Reply:
The first one is what I believe it is.

Reply:
Resource contains the directory itself that we call at the end `addDirectory(resource, callback)`

Reply:
You want `fs.realpath` like in your first commit. It was in the right direction, but async logic was handled incorrectly.

Reply:
Actually realpath is optional property of File System so many tests were failing with `fs.realpath` is not a function.

Reply:
Can you put here test case? I will help

Reply:
Ok, let me try with fs.realpath optionally, if it doesn't exist, it means symlinks are not supported and there's no need for visited logic there. I'll put any test that is failing then.

Reply:
Yes, no tests are failing now.